In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/GiveMeSomeCredit/Data Dictionary.xls
/kaggle/input/GiveMeSomeCredit/cs-training.csv
/kaggle/input/GiveMeSomeCredit/sampleEntry.csv
/kaggle/input/GiveMeSomeCredit/cs-test.csv


<img src="https://i.ibb.co/1T5Wbn6/Banner-para-Linkedin-capa-de-perfil-para-programador.png"  alt="KCode Banner"  />

<a href="https://www.linkedin.com/in/kauebr/"><img src="https://cdn-icons-png.flaticon.com/512/174/174857.png" width="30" ></a> <a href="mailto: kauemandarino@gmail.com"><img src="https://cdn-icons-png.flaticon.com/512/5968/5968534.png" width="30" ></a><a href="https://wa.me/5541991835800?text=Ol%C3%A1%20Kau%C3%AA,%20achei%20seu%20n%C3%BAmero%20no%20GitHub,%20tudo%20bem?"> <img src="https://imagepng.org/wp-content/uploads/2017/08/whatsapp-icone-1.png" width="30" ></a><a href="https://github.com/kauebr/"><img src="https://github.githubassets.com/images/modules/logos_page/Octocat.png" width="30" ></a>

## Initially, I will only calculate through the P-value whether a raised hypothesis is null or alternative.

In [3]:
import pandas as pd

In [9]:
!pip install xlrd

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.5/96.5 kB 3.3 MB/s eta 0:00:00


In [13]:
# Loading and reading our data dictonary

dict = pd.read_excel('/kaggle/input/GiveMeSomeCredit/Data Dictionary.xls')
dict.head(100)

Unnamed: 0  \
0                          Variable Name   
1                       SeriousDlqin2yrs   
2   RevolvingUtilizationOfUnsecuredLines   
3                                    age   
4   NumberOfTime30-59DaysPastDueNotWorse   
5                              DebtRatio   
6                          MonthlyIncome   
7        NumberOfOpenCreditLinesAndLoans   
8                NumberOfTimes90DaysLate   
9           NumberRealEstateLoansOrLines   
10  NumberOfTime60-89DaysPastDueNotWorse   
11                    NumberOfDependents   

                                           Unnamed: 1  Unnamed: 2  
0                                         Description        Type  
1   Person experienced 90 days past due delinquenc...         Y/N  
2   Total balance on credit cards and personal lin...  percentage  
3                            Age of borrower in years     integer  
4   Number of times borrower has been 30-59 days p...     integer  
5   Monthly debt payments, alimony,living costs di...  percentage  
6                                      Monthly income        real  
7   Number of Open loans (installment like car loa...     integer  
8   Number of times borrower has been 90 days or m...     integer  
9   Number of mortgage and real estate loans inclu...     integer  
10  Number of times borrower has been 60-89 days p...     integer  
11  Number of dependents in family excluding thems...     integer

In [20]:
# Loading our datasets

train_df = pd.read_csv('/kaggle/input/GiveMeSomeCredit/cs-training.csv')
test_df = pd.read_csv('/kaggle/input/GiveMeSomeCredit/cs-test.csv')
submissions = pd.read_csv('/kaggle/input/GiveMeSomeCredit/sampleEntry.csv')

In [15]:
submissions

Id  Probability
0            1     0.080807
1            2     0.040719
2            3     0.011968
3            4     0.067640
4            5     0.108264
...        ...          ...
101498  101499     0.045363
101499  101500     0.343775
101500  101501     0.006970
101501  101502     0.121994
101502  101503     0.044248

[101503 rows x 2 columns]

> there's a continous target

In [28]:
# Taking a look at dataset
train_df

Unnamed: 0  SeriousDlqin2yrs  RevolvingUtilizationOfUnsecuredLines  \
0                1                 1                              0.766127   
1                2                 0                              0.957151   
2                3                 0                              0.658180   
3                4                 0                              0.233810   
4                5                 0                              0.907239   
...            ...               ...                                   ...   
149995      149996                 0                              0.040674   
149996      149997                 0                              0.299745   
149997      149998                 0                              0.246044   
149998      149999                 0                              0.000000   
149999      150000                 0                              0.850283   

        age  NumberOfTime30-59DaysPastDueNotWorse    DebtRatio  MonthlyIncome  \
0        45                                     2     0.802982         9120.0   
1        40                                     0     0.121876         2600.0   
2        38                                     1     0.085113         3042.0   
3        30                                     0     0.036050         3300.0   
4        49                                     1     0.024926        63588.0   
...     ...                                   ...          ...            ...   
149995   74                                     0     0.225131         2100.0   
149996   44                                     0     0.716562         5584.0   
149997   58                                     0  3870.000000            NaN   
149998   30                                     0     0.000000         5716.0   
149999   64                                     0     0.249908         8158.0   

        NumberOfOpenCreditLinesAndLoans  NumberOfTimes90DaysLate  \
0                                    13                        0   
1                                     4                        0   
2                                     2                        1   
3                                     5                        0   
4                                     7                        0   
...                                 ...                      ...   
149995                                4                        0   
149996                                4                        0   
149997                               18                        0   
149998                                4                        0   
149999                                8                        0   

        NumberRealEstateLoansOrLines  NumberOfTime60-89DaysPastDueNotWorse  \
0                                  6                                     0   
1                                  0                                     0   
2                                  0                                     0   
3                                  0                                     0   
4                                  1                                     0   
...                              ...                                   ...   
149995                             1                                     0   
149996                             1                                     0   
149997                             1                                     0   
149998                             0                                     0   
149999                             2                                     0   

        NumberOfDependents  
0                      2.0  
1                      1.0  
2                      0.0  
3                      0.0  
4                      0.0  
...                    ...  
149995                 0.0  
149996                 2.0  
149997                 0.0  
149998                 0.0  
149999                 0.

Let's check a hypothesis regarding the number of dependents in a family. My hypothesis is that having a higher number of dependents increases a person's responsibility. Subjectively, this may result in them being more responsible and consequently decreasing their chances of defaulting. This hypothesis is quite relative, considering that, on the other hand, an increase in dependents also raises the person's costs. To determine whether a higher number of dependents increases default rates or not, we will calculate the p-value to either reject or not reject the null hypothesis.

Alternative hypothesis: The more dependents, the lower the risk of default for a customer.
Null hypothesis: There is no relationship between the number of dependents and the risk of default for a customer.

In [36]:
# Check for null values
null_values = train_df['SeriousDlqin2yrs'].isnull().sum()

# Check for blank values
blank_values = (train_df['SeriousDlqin2yrs'] == '').sum()

print("Null values:", null_values)
print("Blank values:", blank_values)


Null values: 0
Blank values: 0


In [37]:
import scipy.stats as stats

# Remove rows with missing values in either column
df_cleaned = train_df.dropna(subset=['SeriousDlqin2yrs', 'NumberOfDependents'])

# Calculate point biserial correlation coefficient and p-value
correlation, p_value = stats.pointbiserialr(df_cleaned['NumberOfDependents'], df_cleaned['SeriousDlqin2yrs'])

print("Point biserial correlation coefficient:", correlation)
print("p-value:", p_value)


Point biserial correlation coefficient: 0.04604794352217033
p-value: 2.113791258958302e-69


Based on the extremely low p-value (2.113791258958302e-69), which is smaller than a commonly used significance level (e.g., 0.05), we can conclude that there is significant statistical evidence to reject the null hypothesis. The null hypothesis states that there is no relationship between the 'SeriousDlqin2yrs' column and the 'NumberOfDependents' column in the 'train_df' dataframe. Therefore, based on the test results, we should reject the null hypothesis and conclude that there is a statistically significant relationship between these two variables. However, it is important to note that the point biserial correlation coefficient of 0.04604794352217033 indicates a very weak correlation, which means that the strength of this relationship is practically insignificant from a practical standpoint.